app/controllers/api/v1/facts_controller.rb

```diff
@@ -0,0 +1,76 @@
+class Api::V1::FactsController < ApplicationController
+    include AuthenticationCheck
+  
+    before_action :is_user_logged_in
+    before_action :set_fact, only: [:show, :update, :destroy]
+    before_action :check_access
+  
+    # GET /members/:member_id/facts
+    def index
+      @member = Member.find(params[:member_id])
+      render json: @member.facts # note that because the facts route is nested inside members
+                               # we return only the facts belonging to that member
+    end
+  
+    # GET /members/:member_id/facts/:id
+    def show
+        if @fact
+          render json: @fact
+        else
+          render json: { error: 'Fact not found' }, status: 404
+        end
+      end
+  
+    # POST /members/:member_id/facts
+    def create
+       @member = Member.find(params[:member_id])
+      @fact = @member.facts.new(fact_params)
+      if @fact.save
+        render json: @fact, status: 201
+      else
+        render json: { error: 
+  "The fact entry could not be created. #{@fact.errors.full_messages.to_sentence}"},
+        status: 400
+      end
+    end
+  
+    # PUT /members/:member_id/facts/:id
+    def update
+        if @fact && check_access
```

Comment:
Because `check_access` is a `before_action`, you don't need it in this method or the `delete` method.